p2p/src/net/types/protocol.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2021-2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    collections::HashSet,
+    fmt::{self, Display, Formatter},
+};
+
+use tap::TapFallible;
+
+use common::primitives::semver::SemVer;
+use logging::log;
+use serialization::{Decode, Encode};
+
+/// Protocol type and version.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash, Encode, Decode)]
+pub struct Protocol {
+    version: SemVer,
+    protocol: ProtocolType,
+}
+
+/// Protocol type.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash, Encode, Decode)]
+pub enum ProtocolType {
+    /// The publish/subscription protocol.
+    Pubsub,
```

Comment:
I think this is inconsistent naming because at least `src/pubsub.rs` uses `PubSub`. I don't mind changing `PubSub` to `Pubsub` but I think it would be better if everything was named the same way.